dandiapi/api/views/serializers.py

```diff
@@ -62,6 +66,15 @@ def get_contact_person(self, dandiset: Dandiset):
 
         return extract_contact_person(latest_version)
 
+    def get_star_count(self, dandiset):
+        return dandiset.star_count
+
+    def get_is_starred(self, dandiset):
+        request = self.context.get('request')
+        if not request or not request.user.is_authenticated:
+            return False
+        return dandiset.is_starred_by(request.user)
```

Comment:
This is inefficient as it leads to a SQL query for every dandiset that's listed (and I can confirm this from local testing). I think something more complex needs to be done, for example, retrieving all of the stars that are associated with the dandisets being listed for one page, and add them to the serialized result.

Reply:
@waxlamp mentioned an "N+1 query problem", which sounds like it's the same or similar to what you are bringing up here. This would also be the case for the contact person of each dandiset, right?

Reply:
Indeed this is an N+1 problem. This doesn't happen with contact person as the `get_contact_person` method that _would_ cause this (originally defined in `DandisetSerializer`) is [overwritten ](https://github.com/dandi/dandi-archive/blob/stars/dandiapi/api/views/serializers.py#L160-L173) in `DandisetListSerializer`, to operate without making extra queries. (Side note, I think this is a bit unnecessary. I don't see a real reason we'd need to display the contact person at the dandiset list level).

Anyway, I think that code snippet is a good example of how we would update this implementation to solve the N+1 query problem, using that `self.context` field.

Reply:
Addressed in [b27390a](https://github.com/dandi/dandi-archive/pull/2123/commits/b27390a07d9952ac06d2343c90aa2954cc9d3f89).